decode.go

```diff
@@ -178,19 +182,44 @@ func findFirstMultiByteChar(data string) int {
 	return len(data)
 }
 
-func (d *Decoder) readLine(v reflect.Value) (err error, ok bool) {
-	line, err := d.data.ReadString('\n')
-	if err != nil && err != io.EOF {
-		return err, false
+func (d *Decoder) SetSeparator(separator string) {
+	d.separator = separator
+}
+
+func (d Decoder) Separator() string {
```

Comment:
I don't see any reason this should be exported. Also is there any reason it is not a pointer receiver e.g. `func (d *Decoder)`?